### PR TITLE
either preload OR inline CSS

### LINKED ISSUE #6008: [v2] CSS preloading observations
## Description
Seeing a Chrome and Lighthouse warning related to CSS assets not being preloaded (correctly?) when using `gatsby serve` after upgrading to latest v2 beta releases.

### Steps to reproduce
Upgraded to latest beta releases of Gatsby v2 as recommended in the v2 beta announcement blog.

### Expected result
No console warning or recommendation from Lighthouse.  (compared to v1)

#### Chrome
screen shot 2018-06-18 at 8 11 36 pm

screen shot 2018-06-18 at 8 13 06 pm

#### Lighthouse
screen shot 2018-06-18 at 8 12 23 pm

### Environment
```shell
$ ./node_modules/.bin/gatsby info --clipboard

  System:
    OS: macOS High Sierra 10.13.5
    CPU: x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.9.4 - /usr/local/bin/node
    Yarn: 1.7.0 - /usr/local/bin/yarn
    npm: 5.6.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 67.0.3396.87
    Firefox: 60.0.1
    Safari: 11.1.1
  npmPackages:
    gatsby: ^2.0.0-beta.1 => 2.0.0-beta.1
    gatsby-cli: ^1.1.58 => 1.1.58
    gatsby-plugin-google-analytics: ^1.0.20-10 => 1.0.20-10
    gatsby-plugin-react-helmet: ^2.0.12-7 => 2.0.12-7
    gatsby-plugin-sitemap: ^1.2.14-10 => 1.2.14-10
    gatsby-plugin-typography: ^2.0.1-11 => 2.0.1-11
```

### File contents (if changed)
`gatsby-config.js`: N/A 
`package.json`
```shell
$ git diff package.json
diff --git a/package.json b/package.json
index efa965a..ca4a395 100644
--- a/package.json
+++ b/package.json
@@ -31,12 +31,12 @@
     "enzyme-adapter-react-16": "^1.1.1",
     "eslint": "^4.19.1",
     "eslint-plugin-react": "^7.7.0",
-    "gatsby": "next",
+    "gatsby": "^2.0.0-beta.1",
     "gatsby-cli": "^1.1.58",
-    "gatsby-plugin-google-analytics": "next",
-    "gatsby-plugin-react-helmet": "next",
-    "gatsby-plugin-sitemap": "next",
-    "gatsby-plugin-typography": "next",
+    "gatsby-plugin-google-analytics": "^1.0.20-10",
+    "gatsby-plugin-react-helmet": "^2.0.12-7",
+    "gatsby-plugin-sitemap": "^1.2.14-10",
+    "gatsby-plugin-typography": "^2.0.1-11",
     "glob": "^7.1.2",
     "jest": "^22.4.3",
     "prop-types": "^15.6.1",
```
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

### PR DESCRIPTION
**Pull request by thescientist13**
Resolves #6008 by either inlining styles OR prefetching them, but not both. 

**Note**: There where modifications to my local _yarh.lock_ file (not committed here) but was wondering if maybe it should be (in a different PR maybe?).  Not sure if maybe a recent contribution missed regenerating the local file?  Here's what I see locally
<details>
```shell
$ git diff yarn.lock
diff --git a/yarn.lock b/yarn.lock
index df3df885..76f10051 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -3033,7 +3033,7 @@ caniuse-db@^1.0.30000529, caniuse-db@^1.0.30000634, caniuse-db@^1.0.30000639:
   version "1.0.30000856"
   resolved "https://registry.yarnpkg.com/caniuse-db/-/caniuse-db-1.0.30000856.tgz#fbebb99abe15a5654fc7747ebb5315bdfde3358f"

-caniuse-lite@^1.0.30000805, caniuse-lite@^1.0.30000844:
+caniuse-lite@^1.0.30000792, caniuse-lite@^1.0.30000805, caniuse-lite@^1.0.30000844:
   version "1.0.30000856"
   resolved "https://registry.yarnpkg.com/caniuse-lite/-/caniuse-lite-1.0.30000856.tgz#ecc16978135a6f219b138991eb62009d25ee8daa"

@@ -3937,7 +3937,7 @@ core-js@^1.0.0:
   version "1.2.7"
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-1.2.7.tgz#652294c14651db28fa93bd2d5ff2983a4f08c636"

-core-js@^2.4.0, core-js@^2.4.1, core-js@^2.5.0, core-js@^2.5.7:
+core-js@^2.4.0, core-js@^2.4.1, core-js@^2.5.0, core-js@^2.5.3, core-js@^2.5.7:
   version "2.5.7"
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-2.5.7.tgz#f972608ff0cead68b841a16a932d0b183791814e"

@@ -4982,7 +4982,7 @@ ejs@^2.4.1:
   version "2.6.1"
   resolved "https://registry.yarnpkg.com/ejs/-/ejs-2.6.1.tgz#498ec0d495655abc6f23cd61868d926464071aa0"

-electron-to-chromium@^1.2.7, electron-to-chromium@^1.3.47:
+electron-to-chromium@^1.2.7, electron-to-chromium@^1.3.30, electron-to-chromium@^1.3.47:
   version "1.3.48"
   resolved "https://registry.yarnpkg.com/electron-to-chromium/-/electron-to-chromium-1.3.48.tgz#d3b0d8593814044e092ece2108fc3ac9aea4b900"

@@ -10673,14 +10673,6 @@ parse-json@^4.0.0:
     error-ex "^1.3.1"
     json-parse-better-errors "^1.0.1"

-parse-latin@^3.1.0:
-  version "3.2.0"
-  resolved "https://registry.yarnpkg.com/parse-latin/-/parse-latin-3.2.0.tgz#cfe4b420982b1d20fc16c71dfb33f148de4f1d0b"
-  dependencies:
-    nlcst-to-string "^2.0.0"
-    unist-util-modify-children "^1.0.0"
-    unist-util-visit-children "^1.0.0"
-
 parse-latin@^4.0.0:
   version "4.1.1"
   resolved "https://registry.yarnpkg.com/parse-latin/-/parse-latin-4.1.1.tgz#3a3edef405b2d5dce417b7157d3d8a5c7cdfab1d"
@@ -12751,11 +12743,11 @@ retext-english@^3.0.0:
     parse-english "^4.0.0"
     unherit "^1.0.4"

-retext-latin@^1.0.0:
-  version "1.0.0"
-  resolved "https://registry.yarnpkg.com/retext-latin/-/retext-latin-1.0.0.tgz#57f41257b4d857b6c6df631ca5788d10080d7051"
+retext-latin@^2.0.0:
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/retext-latin/-/retext-latin-2.0.0.tgz#b11bd6cae9113fa6293022a4527cd707221ac4b6"
   dependencies:
-    parse-latin "^3.1.0"
+    parse-latin "^4.0.0"
     unherit "^1.0.4"

 retext-smartypants@^3.0.0:
@@ -12765,9 +12757,9 @@ retext-smartypants@^3.0.0:
     nlcst-to-string "^2.0.0"
     unist-util-visit "^1.0.0"

-retext-stringify@^1.0.0:
-  version "1.0.0"
-  resolved "https://registry.yarnpkg.com/retext-stringify/-/retext-stringify-1.0.0.tgz#0032e4be3254ea56b19242ef406515479a3346a9"
+retext-stringify@^2.0.0:
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/retext-stringify/-/retext-stringify-2.0.0.tgz#00238facc5491f5bcdc589703a4658db2e54415b"
   dependencies:
     nlcst-to-string "^2.0.0"

@@ -14727,19 +14719,6 @@ unified@^4.1.1:
     trough "^1.0.0"
     vfile "^1.0.0"

-unified@^5.0.0:
-  version "5.1.0"
-  resolved "https://registry.yarnpkg.com/unified/-/unified-5.1.0.tgz#61268da9b91ce925be1f3d198c0278b0e9716094"
-  dependencies:
-    bail "^1.0.0"
-    extend "^3.0.0"
-    has "^1.0.1"
-    is-buffer "^1.1.4"
-    once "^1.3.3"
-    trough "^1.0.0"
-    vfile "^2.0.0"
-    x-is-string "^0.1.0"
-
 unified@^6.0.0, unified@^6.1.4, unified@^6.1.5:
   version "6.2.0"
   resolved "https://registry.yarnpkg.com/unified/-/unified-6.2.0.tgz#7fbd630f719126d67d40c644b7e3f617035f6dba"
```
</details>

### CODE REVIEW

**Comment by thescientist13** [comment]
Note: There where modifications to my local _yarh.lock_ file (not committed here) but was wondering if maybe it should be (in a different PR maybe?).  Not sure if maybe a recent contribution missed regenerating the local file?  Here's what I see locally
```shell
$ git diff yarn.lock
diff --git a/yarn.lock b/yarn.lock
index df3df885..76f10051 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -3033,7 +3033,7 @@ caniuse-db@^1.0.30000529, caniuse-db@^1.0.30000634, caniuse-db@^1.0.30000639:
   version "1.0.30000856"
   resolved "https://registry.yarnpkg.com/caniuse-db/-/caniuse-db-1.0.30000856.tgz#fbebb99abe15a5654fc7747ebb5315bdfde3358f"

-caniuse-lite@^1.0.30000805, caniuse-lite@^1.0.30000844:
+caniuse-lite@^1.0.30000792, caniuse-lite@^1.0.30000805, caniuse-lite@^1.0.30000844:
   version "1.0.30000856"
   resolved "https://registry.yarnpkg.com/caniuse-lite/-/caniuse-lite-1.0.30000856.tgz#ecc16978135a6f219b138991eb62009d25ee8daa"

@@ -3937,7 +3937,7 @@ core-js@^1.0.0:
   version "1.2.7"
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-1.2.7.tgz#652294c14651db28fa93bd2d5ff2983a4f08c636"

-core-js@^2.4.0, core-js@^2.4.1, core-js@^2.5.0, core-js@^2.5.7:
+core-js@^2.4.0, core-js@^2.4.1, core-js@^2.5.0, core-js@^2.5.3, core-js@^2.5.7:
   version "2.5.7"
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-2.5.7.tgz#f972608ff0cead68b841a16a932d0b183791814e"

@@ -4982,7 +4982,7 @@ ejs@^2.4.1:
   version "2.6.1"
   resolved "https://registry.yarnpkg.com/ejs/-/ejs-2.6.1.tgz#498ec0d495655abc6f23cd61868d926464071aa0"

-electron-to-chromium@^1.2.7, electron-to-chromium@^1.3.47:
+electron-to-chromium@^1.2.7, electron-to-chromium@^1.3.30, electron-to-chromium@^1.3.47:
   version "1.3.48"
   resolved "https://registry.yarnpkg.com/electron-to-chromium/-/electron-to-chromium-1.3.48.tgz#d3b0d8593814044e092ece2108fc3ac9aea4b900"

@@ -10673,14 +10673,6 @@ parse-json@^4.0.0:
     error-ex "^1.3.1"
     json-parse-better-errors "^1.0.1"

-parse-latin@^3.1.0:
-  version "3.2.0"
-  resolved "https://registry.yarnpkg.com/parse-latin/-/parse-latin-3.2.0.tgz#cfe4b420982b1d20fc16c71dfb33f148de4f1d0b"
-  dependencies:
-    nlcst-to-string "^2.0.0"
-    unist-util-modify-children "^1.0.0"
-    unist-util-visit-children "^1.0.0"
-
 parse-latin@^4.0.0:
   version "4.1.1"
   resolved "https://registry.yarnpkg.com/parse-latin/-/parse-latin-4.1.1.tgz#3a3edef405b2d5dce417b7157d3d8a5c7cdfab1d"
@@ -12751,11 +12743,11 @@ retext-english@^3.0.0:
     parse-english "^4.0.0"
     unherit "^1.0.4"

-retext-latin@^1.0.0:
-  version "1.0.0"
-  resolved "https://registry.yarnpkg.com/retext-latin/-/retext-latin-1.0.0.tgz#57f41257b4d857b6c6df631ca5788d10080d7051"
+retext-latin@^2.0.0:
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/retext-latin/-/retext-latin-2.0.0.tgz#b11bd6cae9113fa6293022a4527cd707221ac4b6"
   dependencies:
-    parse-latin "^3.1.0"
+    parse-latin "^4.0.0"
     unherit "^1.0.4"

 retext-smartypants@^3.0.0:
@@ -12765,9 +12757,9 @@ retext-smartypants@^3.0.0:
     nlcst-to-string "^2.0.0"
     unist-util-visit "^1.0.0"

-retext-stringify@^1.0.0:
-  version "1.0.0"
-  resolved "https://registry.yarnpkg.com/retext-stringify/-/retext-stringify-1.0.0.tgz#0032e4be3254ea56b19242ef406515479a3346a9"
+retext-stringify@^2.0.0:
+  version "2.0.0"
+  resolved "https://registry.yarnpkg.com/retext-stringify/-/retext-stringify-2.0.0.tgz#00238facc5491f5bcdc589703a4658db2e54415b"
   dependencies:
     nlcst-to-string "^2.0.0"

@@ -14727,19 +14719,6 @@ unified@^4.1.1:
     trough "^1.0.0"
     vfile "^1.0.0"

-unified@^5.0.0:
-  version "5.1.0"
-  resolved "https://registry.yarnpkg.com/unified/-/unified-5.1.0.tgz#61268da9b91ce925be1f3d198c0278b0e9716094"
-  dependencies:
-    bail "^1.0.0"
-    extend "^3.0.0"
-    has "^1.0.1"
-    is-buffer "^1.1.4"
-    once "^1.3.3"
-    trough "^1.0.0"
-    vfile "^2.0.0"
-    x-is-string "^0.1.0"
-
 unified@^6.0.0, unified@^6.1.4, unified@^6.1.5:
   version "6.2.0"
   resolved "https://registry.yarnpkg.com/unified/-/unified-6.2.0.tgz#7fbd630f719126d67d40c644b7e3f617035f6dba"
```

File: packages/gatsby/src/cache-dir/static-entry.js [hunk 1]
```diff
@@ -239,28 +239,32 @@ export default (pagePath, callback) => {
...
+            rel={style.rel}
+            key={style.name}
+            href={urlJoin(pathPrefix, style.name)}
+          />
+        )
+      } else {
+        headComponents.unshift(
+          <style
+            type="text/css"
+            data-href={urlJoin(pathPrefix, style.name)}
```
Comment (thescientist13) [inline packages/gatsby/src/cache-dir/static-entry.js]: Not sure if `data-href` should be removed in this conditional?

Comment (KyleAMathews) [reply packages/gatsby/src/cache-dir/static-entry.js]: No that should be there still. Is there so webpack knows the styles have already been loaded so won't try again.

**Comment by m-allanson** [comment]
A separate PR for updating the yarn.lock file would be splendid 👌, if you get a chance to do that.

**Comment by thescientist13** [comment]
@m-allanson / @KyleAMathews 
Based on the [build failure](https://ci.appveyor.com/project/KyleAMathews/gatsby/build/1.0.7936/job/g9nwewycw6lfswni), should I update Jest snapshots?

**Comment by thescientist13** [comment]
@m-allanson 
Looks like the _yarn.lock_ file was updated as part of [this PR](https://github.com/gatsbyjs/gatsby/pull/6002).  Running `yarn bootstrap` on latest version of the v2 branch shows now diffs on _yarn.lock_, so that's all set now!  🎉

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by KyleAMathews** [comment]
Just ended up cherry picking this change :-) Gotta kill that v2 branch later haha

**Comment by thescientist13** [comment]
congrats on the merge of v2 into master!!  🎉

**Comment by thescientist13** [comment]
@m-allanson 
I don't have a windows machine, but I am comfortable with setting up Windows via a VM (like VirtualBox).  Would that be sufficient to reproduce #5725 ?

**Comment by thescientist13** [comment]
oh, looks like it may have already been solved?  👀 

your team is quick!  🏃

**Pull request merged**